### ISSUE TITLE
%python_uninstall_alternative includes updates-alternatives --remove

### ISSUE THREAD

**Issue by mcepl** [rpm-software-management/rpmlint#695]
With [python-blockdiag.spec](https://build.opensuse.org/package/view_file/devel:languages:python/python-blockdiag/python-blockdiag.spec?expand=1) we have this in scriptlets:
```rpm
%install
%python_install
%python_expand %fdupes %{buildroot}%{$python_sitelib}
%python_clone -a %{buildroot}%{_bindir}/blockdiag

%post
%python_install_alternative blockdiag

%preun
%python_uninstall_alternative blockdiag
```
and yet rpmlint ends with this warnings:
```
[  139s] python38-blockdiag.noarch: E: update-alternatives-postun-call-missing
```
I don't think rpmlint recognizes that `%python_uninstall_alternative` includes `updates-alternatives --remove`.

**Comment by Conan-Kudo** [comment]
That sounds like the macro isn't expanding properly or the result is slightly too obfuscated. What is the actual scriptlet from the built RPM? That is, using `rpm -qp --scripts`?

**Comment by marxin** [comment]
```
rpm -qp --scripts python39-blockdiag-2.0.1-21.1.noarch.rpm
warning: python39-blockdiag-2.0.1-21.1.noarch.rpm: Header V3 DSA/SHA1 Signature, key ID edf0d733: NOKEY
postinstall scriptlet (using /bin/sh):
 
update-alternatives --quiet --install /usr/bin/blockdiag blockdiag /usr/bin/blockdiag-3.9 39
preuninstall scriptlet (using /bin/sh):


if [ ! -e "/usr/bin/blockdiag-3.9" ]; then 
    update-alternatives --quiet --remove "blockdiag" "/usr/bin/blockdiag-3.9" 
fi
marxin@kettlebell:~/Downloads> rpm --qp --scripts python39-blockdiag-2.0.1-21.1.noarch.rpm mb:/tmp/^C
```

which seems correct, but
```
        self.post = byte_to_string(pkg.header[rpm.RPMTAG_POSTIN])
        self.postun = byte_to_string(pkg.header[rpm.RPMTAG_POSTUN])
        print(self.post)
        print(self.postun)
```

prints
```
 
update-alternatives --quiet --install /usr/bin/blockdiag blockdiag /usr/bin/blockdiag-3.9 39
None
```

I'm investigating that more now.

**Comment by marxin** [comment]
> %preun

No, rpmlint is correct, it's checking `postuninstall` and not `preuninstall`.
